server/controllers/rideRequestController.js

```diff
@@ -78,4 +87,62 @@ export default class RideRequest {
         });
       });
   }
+
+  /**
+   * accept or reject a ride request
+   *
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @memberof RideRequest
+   */
+  static respondToRequests(req, res) {
+    const { status } = req.body;
+    const { requestId, rideId } = req.params;
+    const { userId } = req;
+    // check if the user is the one who created the ride offer
+    pool
+      .query('SELECT * FROM rides WHERE id = $1', [rideId])
+      .then((result) => {
+        if (result.rowCount) {
+          if (result.rows[0].userid !== userId) {
+            return res.status(400).json({
+              status: 'error',
+              message: 'You are not allowed to respond to another users ride requests',
+            });
+          }
+          return pool
+            .query('UPDATE requests SET offerstatus = $1, updated_at = NOW() WHERE id = $2', [status, requestId])
+            .then((updateResult) => {
+              if (updateResult.rowCount) {
+                return res.status(200).json({
+                  status: 'success',
+                  message: `Successfully ${status} ride request`,
+                });
+              }
+              return res.status(500).json({
+                status: 'error',
+                message: 'Unable to respond to ride request',
+              });
+            })
+            .catch(() => {
+              return res.status(500).json({
+                status: 'error',
+                message: 'Unable to respond to ride request',
+              });
+            });
+        }
+        return res.status(404).json({
+          status: 'error',
+          message: 'The requested ride was not found',
+        });
+      })
+      .catch((error) => {
+        // could not retrieve ride
+        return res.status(500).json({
+          status: 'error',
+          message: error,
```

Comment:
You shouldn't send auto-generated errors to the consumer of your api because you have no control over what the error message would be and the consumer of your api might rely on its consistency.